eth/client.go

```diff
@@ -738,8 +739,109 @@ func (c *client) Vote(pollAddr ethcommon.Address, choiceID *big.Int) (*types.Tra
 	return poll.Vote(opts, choiceID)
 }
 
+func (c *client) Reward() (*types.Transaction, error) {
+	addr := c.accountManager.Account().Address
+
+	t, err := c.GetTranscoder(addr)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder %v", addr.Hex())
+	}
+
+	mintable, err := c.CurrentMintableTokens()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get curernt mintable tokens")
+	}
+
+	totalBonded, err := c.GetTotalBonded()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get total bonded")
+	}
+
+	// reward = (current mintable tokens for the round * active transcoder stake) / total active stake
+	reward := new(big.Int).Div(new(big.Int).Mul(mintable, t.DelegatedStake), totalBonded)
+
+	hints, err := c.simulateTranscoderPoolUpdate(addr, reward.Add(reward, t.DelegatedStake))
+	if err != nil {
+		glog.Errorf("Unable to calculate hints, submitting reward transaction without hints. err=%v", err)
+	}
+	return c.RewardWithHint(hints.PosPrev, hints.PosNext)
+
+}
+
 // Helpers
 
+// simulateTranscoderPoolUpdate simulated an  update to the transcoder pool and returns the positional hints for a delegator accordingly.
+// if the delegator will not be in the updated set 'nil' will be returned.
+func (c *client) simulateTranscoderPoolUpdate(del ethcommon.Address, newStake *big.Int) (lpTypes.StakingHints, error) {
+
+	// get max pool size
+	maxSize, err := c.GetTranscoderPoolMaxSize()
+	if err != nil {
+		return lpTypes.StakingHints{}, err
+	}
+	maxSizeInt := int(maxSize.Int64())
+
+	// get the transcoder  pool
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return lpTypes.StakingHints{}, err
+	}
+
+	// if the transcoder pool is full and the newStake for 'del' is lower  than that of the last transcoders
+	// 'del' will not be in the active set and thus there are no hints to return.
+	if len(transcoders) == maxSizeInt && newStake.Cmp(transcoders[maxSizeInt-1].DelegatedStake) <= 0 {
```

Comment:
This condition would also evaluate to true when the stake of a transcoder that is already in the pool decreases (i.e. due to unbonding) - we should still generate a hint in that case.

Reply:
I don't think this conditional returns true if the transcoder remains in the transcoder pool after unbonding ? 

The list is full and the updated stake is lower than that of the last transcoder in the list -> transcoder is evicted so there are no hints to be calculated. 

This statement kind of short-circuits this function to skip sorting and a linear search but removing it would lead to the same result.

Reply:
If the pool is full and the stake of a transcoder in the pool decreases such that it is lower than that of the last transcoder in the pool the transcoder would remain in the pool, but it would just be in the last position in the pool. At the moment, I believe this conditional would return true and no hint would be returned when instead we would want to return a hint that indicates that the transcoder should now be in the last position of the pool.

Reply:
Sorry, yup you're right !

Reply:
added a test for this case f79e4f96